src/mca/bfrops/base/bfrop_base_pack.c

```diff
@@ -934,7 +934,7 @@ pmix_status_t pmix_bfrops_base_pack_query(pmix_pointer_array_t *regtypes,
     for (i=0; i < num_vals; i++) {
         /* pack the number of keys */
         nkeys = pmix_argv_count(pq[i].keys);
-        PMIX_BFROPS_PACK_TYPE(ret, buffer, &nkeys, 1, PMIX_UINT32, regtypes);
+        PMIX_BFROPS_PACK_TYPE(ret, buffer, &nkeys, 1, PMIX_INT32, regtypes);
```

Comment:
@rhc54 @jjhursey 
This issue was uncovered during a new flexible integer packing implementation.
The unpacking counterpart is expecting Signed integer.
In order to have an efficient packing for both signed and unsigned integers we pack them differently (because of the internal representation)

The side effect of this is that you have to use exactly the same type for packing and unpacking (signed and unsigned int's are not compatible anymore).
Is it ok with respect to the standard?

Reply:
I apologize for the delayed response - I had missed this question in my email. The standard actually requires that the pack and unpack have the same type. In the library, it will return an error if you pack an unsigned int and attempt to unpack it as an int - we check that the two pmix_data_type_t values match.

Reply:
@rhc54 great! Then we are ok here.
@karasevb is wrapping up my final comments. PRs will be ready for the community till tomorrow.